File: MyHomeSPA/src/app/shoppingPlanners/models/purchase.ts
export class Purchase {

    public id: number;

    public name: string;

    public description: string;

    public typeOfPurchaseId: number;

    public amount: number;

    public priceOfOneUnit: number;

    public totalPrice: number;

    public isDone: boolean;

    public shoppingCategoryId: number;

    constructor() {

        this.id = 0;

        this.name='';

        this.description = '';

        this.typeOfPurchaseId = 0;

        this.amount = 1;

        this.priceOfOneUnit = 1;

        this.totalPrice = this.amount*this.priceOfOneUnit;

        this.isDone = false;

        this.typeOfPurchaseId = 0;

        this.shoppingCategoryId = 0;
    }

    public getTotalPrice(): number{

        this.totalPrice = this.priceOfOneUnit * this.amount;

        return this.totalPrice;
    }
}

File: Common/Entity/ShoppingPlannerService/Purchase.cs
using System.ComponentModel.DataAnnotations;

namespace Common.Entity.ShoppingPlannerService
{
    public class Purchase
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public int TypeOfPurchaseId { get; set; }

        [Required]
        public float Amount { get; set; }

        [Required]
        public float PriceOfOneUnit { get; set; }

        [Required]
        public float TotalPrice { get; set; }

        public bool IsDone { get; set; }

        public int ShoppingCategoryId { get; set; }

        public int PlannerDateId { get; set; }
    }
}
